model/task/task.go

```diff
@@ -464,11 +466,17 @@ func (t *Task) DependenciesMet(depCaches map[string]Task) (bool, error) {
 		if !t.SatisfiesDependency(&depTask) {
 			return false, nil
 		}
-		if depTask.FinishTime.After(latestTime) {
-			latestTime = depTask.FinishTime
-		}
 	}
-	t.DependenciesMetTime = latestTime
+	// this is not exact, but depTask.FinishTime is not always set in time to use that
```

Comment:
Could you check if FinishTime is set and if it's not then use time.Now()?

Reply:
In practice, DependenciesMet is called four times a second, so we should be pretty accurate even if it's not an exact match. Every time I tested it on staging, DependenciesMet returned before FinishTime could be set, so I think we would end up rarely, if ever, getting any benefit. However, I wanted to leave the comment in case anyone is wondering about some slight discrepancy in the future.

Reply:
That makes sense! Wow haha 4 times a second.

Reply:
It's not scheduled or anything, but the last two I ran caught it in about 100 milliseconds. I'm definitely going to keep an eye out once we deploy to make sure this doesn't degrade, but DependenciesMet should be getting called all the time.